lib/sycamore/sycamore/docset.py

```diff
@@ -975,28 +980,48 @@ def llm_filter(
             prompt: LLM prompt.
             field: Document field to filter based on.
             threshold: Cutoff that determines whether or not to keep document.
+            ignore_doc_structure: ignores document.element mapping.
+            similarity_query: query string to compute similarity against.
+            similarity_scorer: scorer to generate similarity if 'sort_elements_by_similarity' is True.
             **resource_args
 
         Returns:
             A filtered DocSet.
         """
-
-        def threshold_filter(doc: Document, threshold) -> bool:
-            try:
-                return_value = int(doc.properties[new_field]) >= threshold
-            except Exception:
-                # accounts for llm output errors
-                return_value = False
-
-            return return_value
-
-        docset = self.filter(lambda doc: doc.field_to_value(field) is not None and doc.field_to_value(field) != "None")
-
         entity_extractor = OpenAIEntityExtractor(
             entity_name=new_field, llm=llm, use_elements=False, prompt=prompt, field=field
         )
-        docset = docset.extract_entity(entity_extractor=entity_extractor)
-        docset = docset.filter(lambda doc: threshold_filter(doc, threshold), **resource_args)
+
+        def threshold_filter(doc: Document, threshold) -> bool:
+            if ignore_doc_structure:
+                if doc.field_to_value(field) is None:
+                    return False
+                doc = entity_extractor.extract_entity(doc)
+                return int(re.findall(r"\d+", doc.properties[new_field])[0]) >= threshold
```

Comment:
This is a strange operation. You're just returning the integer value of the first string of digits, but why? That seems incredibly brittle. It seems to me we want to enforce that the LLM output of the entity extractor is a numeric value in a certain range, not try to hack around it here if it's not. Can we push some logic down to `entity_extractor` with a validator to ensure the result is something we expect, and set it to, e.g., None, if not?

If you're trying to avoid brittleness in the entity_extractor here, you should also be using `doc.properties.get(new_field)` and using a default value in case it did not populate the field at all.

Reply:
>. Can we push some logic down to entity_extractor with a validator to ensure the result is something we expect, and set it to, e.g., None, if not?
This is the right solution but I don't want to combine it in this PR, there are quiet a few places we need to update. 

>If you're trying to avoid brittleness in the entity_extractor here, you should also be using doc.properties.get(new_field) and using a default value in case it did not populate the field at all.
We had discussed this earlier in the error handling discussion where we don't want to put missing data handling in operations (I believe it was missing dates from ntsb data), staying consistent with that for now. I can add a ignore_nulls style flag though